sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala

```diff
@@ -121,6 +134,25 @@ private[hive] object SparkSQLCLIDriver extends Logging {
       }
     }
 
+    if (isSecuredAndProxy(conf)) {
+      val currentUser = UserGroupInformation.getCurrentUser
+      try {
+        SparkHadoopUtil.get.doAsRealUser {
```

Comment:
Couldn't you just call `HiveDelegationTokenProvider` here instead of copy & pasting the code?

Reply:
updated